src/rust/engine/remote_provider/remote_provider_opendal/src/lib.rs

```diff
@@ -173,16 +173,14 @@ impl Provider {
             Err(e) => return Err(format!("failed to read {}: {}", path, e)),
         };
 
-        if let Some(workunit_store_handle) = workunit_store::get_workunit_store_handle() {
-            // TODO: this pretends that the time-to-first-byte can be approximated by "time to create
-            // reader", which is often not really true.
-            let timing: Result<u64, _> =
-                Instant::now().duration_since(start).as_micros().try_into();
-            if let Ok(obs) = timing {
-                workunit_store_handle
-                    .store
-                    .record_observation(ObservationMetric::RemoteStoreTimeToFirstByteMicros, obs);
-            }
+        // TODO: this pretends that the time-to-first-byte can be approximated by "time to create
+        // reader", which is often not really true.
+        let timing: Result<u64, _> = Instant::now().duration_since(start).as_micros().try_into();
```

Comment:
In some cases this might now "unnecessarily" do this time computation (if not running in a work unit), but I think that's rare, so that's fine?

Reply:
Seems fine by me. My (untested) educated guess is that this code will run more often in a work unit than not.

Reply:
A more complex helper would take a `FnOnce() -> u64` closure but that seems unnecessary right now. E.g., `increment_counter_if_in_workunit_with`.